src/xcode/ENA/ENA/Source/Coordinator.swift

```diff
@@ -84,7 +84,17 @@ class Coordinator: RequiresAppDependencies {
 			#endif
 		})
 	}
-
+	
+	func showPositiveTestResultFromNotification(with result: TestResult) {
+		guard ((rootViewController.presentedViewController?.dismiss(animated: true, completion: {
+			self.showExposureSubmission(with: result)
+		})) != nil) else {
+			self.showExposureSubmission(with: result)
+			return
+		}
```

Comment:
I find this a little hard to read, maybe you can make this a bit more explicit like this?
```
if let presentedViewController = rootViewController.presentedViewController {
	presentedViewController.dismiss(animated: true) {
		self.showExposureSubmission(with: result)
	}
} else {
	self.showExposureSubmission(with: result)
}
```

Reply:
Thanks Nick, I also think like this before😉. It's done now.